packages/flutter/lib/src/rendering/box.dart

```diff
@@ -1945,7 +1945,8 @@ abstract class RenderBox extends RenderObject {
         if (RenderObject.debugActiveLayout != null) {
           assert(
             debugDoingThisResize || debugDoingThisLayout || _computingThisDryLayout ||
-              (RenderObject.debugActiveLayout == parent && _size._canBeUsedByParent),
+              (RenderObject.debugActiveLayout == parent && _size._canBeUsedByParent) ||
+                RenderObject.debugActiveLayout!.debugDoingThisLayoutWithCallback,
```

Comment:
can you move this into the if case in line 1945? This is more like something we want to short-circuit than checking in assert

Reply:
done